service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/RegistryManager.java

```diff
@@ -59,15 +66,20 @@ public RegistryManager()
                 .httpConnectTimeout(RegistryManagerOptions.DEFAULT_HTTP_CONNECT_TIMEOUT_MS)
                 .httpReadTimeout(RegistryManagerOptions.DEFAULT_HTTP_READ_TIMEOUT_MS)
                 .build();
+        hostName = ""; // only so that hostName can be declared final
     }
 
     /**
      * Static constructor to create instance from connection string
      *
      * @param connectionString The iot hub connection string
      * @return The instance of RegistryManager
-     * @throws IOException This exception is thrown if the object creation failed
+     * @deprecated because this method declares a thrown IOException even though it never throws an IOException. Users
+     * are recommended to use {@link #RegistryManager(String)} instead since it does not declare this exception even
+     * though it constructs the same RegistryManager.
+     * @throws IOException This exception is never thrown.
```

Comment:
So the reason why we are deprecating this is because it never throws an exception as noted in documentation?

Reply:
I believe Tim investigated, and an IOException is never thrown. Maybe it did at some point, but that's no longer the case.

Reply:
Yep. I created a non-static constructor that takes the same arguments which will replace this static constructor